src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -83,21 +101,49 @@ public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBu
      */
     public ServiceBusJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
         this.settings = settings;
-        this.initialize(sasKeyName, sasKey, host, settings);
+        this.setKeyValue(sasKey);
+        this.keyName = sasKeyName;
+        this.initialize(host, settings);
     }
     
-    private void initialize(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
-        if (sasKeyName == null || sasKeyName == null || host == null) {
-            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
-        }
-        
+    /**
+     * Create a ServiceBusJmsConnectionFactory using shared access key and host name.
+     * @param tenantId Tenent id 
+     * @param clientId Client id.
+     * @param clientSecret Client secret
+     * @param host The host name of the ServiceBus namespace. Example: your-namespace-name.servicebus.windows.net
+     * @param settings The options used for this ConnectionFactory. Null can be used as default.
+     */
+    public ServiceBusJmsConnectionFactory(String tenantId, String clientId, String clientSecret, String host, ServiceBusJmsConnectionFactorySettings settings) {
+        this.settings = settings;
+        this.setIsToken(true);
```

Comment:
>setIsToken [](http://example.com/codeflow?start=13&length=10)

nit: function name can be setIsAadToken()

Reply:
Ths is not a validation of if is aad token just if the user is going to provide they own token and it was add since it had some interfaces with the tokenprovider constructor. Will be remove since this facture is no going to be supported.

Reply:
Correction I change it to  isAadAuthentication since we  have some extensions done for JNDI were we need to do if statements. Talk with shankar about possibility of having a separation using abstract but we may end up breaking existent customers. Something we can explore